projects/components/src/overlay/modal/modal-ref.ts

```diff
@@ -0,0 +1,14 @@
+import { Subject } from 'rxjs';
+import { PopoverRef } from '../../popover/popover-ref';
+
+export class ModalRef extends PopoverRef {
+  public readonly output$: Subject<unknown> = new Subject<unknown>();
```

Comment:
I would argue for keeping this completely decoupled from popover rather than extending it - it's built on top of popover, but should be completely its own concept.

Reply:
What does that gain us here when we just want one extra property?

Reply:
It's not just one extra property though - most of the methods on popoverRef are not relevant to modalRef. In other words, inheritance is a bad fit here. What methods do we actually want to expose on ModalRef? I'd think something along the lines of below to start.
```typescript
interface ModalRef<TResult> {
   closed$: Observable<TResult>;
   close(result: TResult): void;
}
```
As far as implementation, we can easily take in a popoverref to the ModalRef implementation to support this. A good rule of thumb on inheritance - are you actually using it polymorphically? If yes, and it satisfies an `is a` relationship, then it might be a good fit. If it's being used to cut down on code written, it's likely not. It tightly couples things, making the child much more susceptible to changes in the parent (vs changes in the parent's API in the suggested wrapping approach) and makes it harder to test in isolation as you can't separate the parent logic out to mock it.

Reply:
I would agree with the api aaron suggested. If we use has-a relationship, it would be cleaner and it would avoid unnecessary changes to popover to support the inheritance.